File: jphp-javafx-ext/src/main/resources/JPHP-INF/sdk/ext/javafx/event/UXContextMenuEvent.php
<?php
namespace ext\javafx\event;

/**
 * Class UXContextMenuEvent
 * @package ext\javafx\event
 */
class UXContextMenuEvent extends UXEvent
{
    /**
     * @readonly
     * @var double
     */
    public $sceneX;

    /**
     * @readonly
     * @var double
     */
    public $sceneY;

    /**
     * @readonly
     * @var double
     */
    public $screenX;

    /**
     * @readonly
     * @var double
     */
    public $screenY;

    /**
     * @readonly
     * @var double
     */
    public $x;

    /**
     * @readonly
     * @var double
     */
    public $y;

    /**
     * @readonly
     * @var bool
     */
    public $keyboardTrigger;
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/event/UXContextMenuEvent.java
package org.develnext.jphp.ext.javafx.classes.event;

import javafx.scene.input.ContextMenuEvent;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "event\\UXContextMenuEvent")
public class UXContextMenuEvent extends UXEvent {
    interface WrappedInterface {
        @Property double sceneX();
        @Property double sceneY();

        @Property double screenX();
        @Property double screenY();

        @Property double x();
        @Property double y();

        @Property boolean keyboardTrigger();
    }

    public UXContextMenuEvent(Environment env, ContextMenuEvent wrappedObject) {
        super(env, wrappedObject);
    }

    public UXContextMenuEvent(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public ContextMenuEvent getWrappedObject() {
        return (ContextMenuEvent) super.getWrappedObject();
    }
}
